node/sync/src/block_sync.rs

```diff
@@ -381,11 +381,11 @@ impl<N: Network> BlockSync<N> {
 
         // Compute the common ancestor with this node.
         let mut ancestor = 0;
-        for (height, hash) in locators.clone().into_iter() {
+        for (height, hash) in locators.clone().into_iter().rev() {
             if let Ok(canon_hash) = self.canon.get_block_hash(height) {
-                match canon_hash == hash {
-                    true => ancestor = height,
-                    false => break, // fork
+                if canon_hash == hash {
+                    ancestor = height;
+                    break;
```

Comment:
@ljedrz can you keep the original logic but encapsulate it with rayon (to check each iteration of the look in parallel)? We want to do fork detection, which is what the original for loop does. Please add a comment to denote this for future readers not to remove.

Then, in addition, you can add the new early-return for loop as a way to quickly detect the latest common ancestor.

Reply:
> can you keep the original logic but encapsulate it with rayon

Unfortunately, this would be tricky due to the reassignment of the `ancestor` and the breaking logic. However, the addition of the loop skip means it's no longer a performance concern, as forks are unusual.

> add a comment to denote this for future readers not to remove.

Done.

> add the new early-return for loop as a way to quickly detect the latest common ancestor.

Done.